src/primes.rs

```diff
@@ -88,8 +89,8 @@ impl fmt::Display for Edn {
 }
 
 fn is_simple_token(tok: &str) -> bool {
-  for s in tok.chars() {
-    if !matches!(s, '0'..='9' | 'A'..='Z'| 'a'..='z'|  '-' | '?' | '.'| '$' | ',') {
+  for s in tok.bytes() {
+    if !matches!(s as char, '0'..='9' | 'A'..='Z'| 'a'..='z'|  '-' | '?' | '.'| '$' | ',') {
```

Comment:
```suggestion
    if !matches!(s, b'0'..=b'9' | b'A'..=b'Z'| b'a'..=b'z'|  b'-' | b'?' | b'.'| b'$' | b',') {
```
use this directly should be more performant I guess, It need not convert `u8` to `char`, which using `u32` under the hood.

Reply:
thanks